### PR TITLE
Add command for remove images without reference object

### PR DESCRIPTION
**Pull request by valiknet18**
ticket: https://app.asana.com/0/25002076043610/29743393245510

### CODE REVIEW

File: src/AppBundle/Command/RemoveImageCommand.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        ;
+
+        $medias = $mm->findAll();
+
+        foreach ($medias as $media) {
+            $context = $media->getContext() == "slider" ? "performance" : $media->getContext();
+            $entity = "AppBundle\\Entity\\".ucfirst($context);
+
+            switch ($media->getContext()) {
+                case 'employee':
```
Comment (spolischook) [inline src/AppBundle/Command/RemoveImageCommand.php]: So. What will happen if I'll add/edit/delete context? I'll need to edit command, right? 
Move this logic into separate method, and rewrite it all.

File: src/AppBundle/Command/RemoveImageCommand.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $mm = $this
+            ->getContainer()
+            ->get('sonata.media.manager.media')
+        ;
+
+        $em = $this
```
Comment (spolischook) [inline src/AppBundle/Command/RemoveImageCommand.php]: I think you can to delete MediaObject without EntityManager - you need only MediaManager - use it.

File: src/AppBundle/Command/RemoveImageCommand.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace AppBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+
+class RemoveImageCommand extends ContainerAwareCommand
```
Comment (spolischook) [inline src/AppBundle/Command/RemoveImageCommand.php]: RemoveBrokenMediaObjectsCommand

File: src/AppBundle/Tests/Command/RemoveImageCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+namespace AppBundle\Tests\Command;
+
+use AppBundle\Command\RemoveImageCommand;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+
+class RemoveImageCommandTest extends KernelTestCase
+{
+    public function testExecuteCommand()
```
Comment (spolischook) [inline src/AppBundle/Tests/Command/RemoveImageCommandTest.php]: This test - test only that command will execute command without Exception.
You can write functional test:
 - Check that db does not have broken MediaObjects
 - Write broken MediaObject
 - Check
 - execute a command
 - Check
Also, after refactoring RemoveBrokenMediaObjectsCommand:execute and devide it to some separate methods you can unit tested protected methods.

File: src/AppBundle/Command/RemoveImageCommand.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+{
+    protected function configure()
+    {
+        $this
+            ->setName('app:remove-image')
+            ->setDescription('Remove image that not relation to media')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (spolischook) [inline src/AppBundle/Command/RemoveImageCommand.php]: Too large method. Please, devide it into separate protected methods.

File: src/AppBundle/Command/RemoveBrokenMediaObjectsCommand.php [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+        return $property;
+    }
+
+    /**
+     * @param $media
+     * @param $objects
+     * @return int
+     */
+    protected function proccessMediaObject($media, $objects)
```
Comment (spolischook) [inline src/AppBundle/Command/RemoveBrokenMediaObjectsCommand.php]: proccessMediaObject - this name say me nothing about method content

File: src/AppBundle/Command/RemoveBrokenMediaObjectsCommand.php [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        $em = $this
+            ->getContainer()
+            ->get('doctrine')
+            ->getManager()
+        ;
+
+        $medias = $mm->findAll();
+
+        foreach ($medias as $media) {
+            $context = $media->getContext() == "slider" ? "performance" : $media->getContext();
```
Comment (spolischook) [inline src/AppBundle/Command/RemoveBrokenMediaObjectsCommand.php]: And what if I add of edit current contexts? I'll need to change command? It's not good.

File: src/AppBundle/Command/RemoveBrokenMediaObjectsCommand.php [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+     * @return int|null|void
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $mm = $this
+            ->getContainer()
+            ->get('sonata.media.manager.media')
+        ;
+
+        $em = $this
```
Comment (spolischook) [inline src/AppBundle/Command/RemoveBrokenMediaObjectsCommand.php]: if you need you can get it like that:
```php
$mm->getEntityManager()
```
but I still sure that you don't need it

**Comment by mushkin-v** [comment]
LGTM

**Comment by iboved** [comment]
LGTM

**Comment by Slavka-JD** [comment]
LGTM

**Pull request merged**